packages/next/src/server-build.ts

```diff
@@ -1090,6 +1090,36 @@ export async function serverBuild({
     )
   );
 
+  const pagesPlaceholderRscEntries: Record<string, FileFsRef> = {};
+
+  if (appDir) {
+    // since we attempt to rewrite all paths to an .rsc variant,
+    // we need to create dummy rsc outputs for all pages entries
+    // this is so that an RSC request to a `pages` entry will match
+    // rather than falling back to a catchall `app` entry
+    // on the nextjs side, invalid RSC response payloads will correctly trigger an mpa navigation
+    const pagesManifest = path.join(
+      entryPath,
+      outputDirectory,
+      `server/pages-manifest.json`
+    );
+
+    const pagesData = await fs.readJSON(pagesManifest);
+    const pagesEntries = Object.keys(pagesData);
+
+    if (pagesEntries.length > 0) {
+      const placeholderFilePath = path.join(tmpdir(), 'rsc-placeholder');
+      await fs.writeFile(placeholderFilePath, 'RSC Placeholder');
+
+      for (const page of pagesEntries) {
+        const pathName = page.startsWith('/') ? page.slice(1) : page;
+        pagesPlaceholderRscEntries[`${pathName}.rsc`] = new FileFsRef({
```

Comment:
Do we also need the dummy `.prefetch.rsc` outputs as that is being requested when the prefetch header is present?